subiquity/server/controllers/mirror.py

```diff
@@ -130,8 +141,16 @@ async def try_mirror_checking_once(self) -> None:
             for line in output.getvalue().splitlines():
                 log.debug("%s", line)
 
-    @with_context()
-    async def apply_autoinstall_config(self, context):
+    async def find_and_elect_candidate_mirror(self, context):
+        # This function assumes that the source and proxy models have
+        # already been marked configured. Currently apply_autoinstall_config
+        # coroutines are called sequentially so it should be okay for a fully
+        # non-interactive install. If we were to invoke the coroutines
+        # concurrently, we would need to wait for the source and proxy to
+        # finish first.
+        # TODO check if the same assumption applies in a partially interactive
```

Comment:
I don't think it does. I guess you get to confirm this though!

Reply:
Just as you thought, we can't make the same assumptions. I tested with "proxy" as an interactive section and the mirror testing certainly started way before I had a chance to make "proxy" configured. I added an `await self.proxy_configured.wait()` statement and it works better now.